buildSrc/buildWebapp.js

```diff
@@ -0,0 +1,204 @@
+/**
+	Exports the buildWebapp function that can be used for production builds.
+ */
+import {rollup} from "rollup"
+import typescript from "@rollup/plugin-typescript"
+import {terser} from "rollup-plugin-terser"
+import path from "path"
+import {nodeResolve} from "@rollup/plugin-node-resolve"
+import commonjs from "@rollup/plugin-commonjs"
+import fs from "fs-extra"
+import {bundleDependencyCheckPlugin, getChunkName, resolveLibs} from "./RollupConfig.js"
+import {visualizer} from "rollup-plugin-visualizer"
+import os from "os"
+import * as env from "./env.js"
+import {createHtml} from "./createHtml.js"
+
+/**
+ * Builds the web app for production.
+ * @param version Version of the app. Will be used for html generation and service worker versioning.
+ * @param stage Deployment for which to build: 'prod' will build for the production system, 'test' for the test system, 'local' will use localhost.
+ * @param host If stage is left undefined, the value provided here will be used to construct the app URL for HTML creation.
+ * @param measure Function that returns the current elapsed build time.
+ * @param minify Boolean. Set to true to perform minification.
+ * @param projectDir Path to the tutanota root directory.
+ * @returns Nothing meaningful.
+ */
+
+export async function buildWebapp({version, stage, host, measure, minify, projectDir}) {
```

Comment:
Are there any changes here to note or is it just 1:1 copied? It's hard to tell...

Reply:
This has mostly been extracted into it's own file and had some documentation added. The code itself hardly changed.